lib/open_api_spex/cast/discriminator.ex

```diff
@@ -40,7 +40,7 @@ defmodule OpenApiSpex.Cast.Discriminator do
   defp cast_discriminator(%_{value: value, schema: schema} = ctx) do
     {discriminator_property, mappings} = discriminator_details(schema)
 
-    case value["#{discriminator_property}"] || value[discriminator_property] do
+    case value["#{discriminator_property}"]  do
```

Comment:
@mbuhot I don't remember if I was the one that originally added this line but I think this was needed to support discriminators where the property is an atom, it is indirectly tested in the test you removed in this same commit.

Reply:
I think there are use cases but I don't have a particular one in mind.

My reasoning was that since it is possible to cast an object with atom keys it should be possible also when the discriminator property is an atom

Reply:
@igormq Is there a scenario you have where you need to cast data that already has atom keys?
I'm open to including the functionality, I just want to make sure it's not a hack to fix some other bug we might have :) 

https://github.com/open-api-spex/open_api_spex/pull/440/commits/024e7d206e870ffb8300f194ec6f2ac4b9c46c8a